src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -816,21 +816,22 @@ public abstract partial class JsonSerializerContext
         public abstract System.Text.Json.Serialization.Metadata.JsonTypeInfo? GetTypeInfo(System.Type type);
     }
     [System.AttributeUsageAttribute(System.AttributeTargets.Class, AllowMultiple=false)]
-    public partial class JsonSerializerOptionsAttribute : System.Text.Json.Serialization.JsonAttribute
+    public partial class JsonSourceGenerationOptionsAttribute : System.Text.Json.Serialization.JsonAttribute
     {
-        public JsonSerializerOptionsAttribute() { }
+        public JsonSourceGenerationOptionsAttribute() { }
         public System.Text.Json.Serialization.JsonIgnoreCondition DefaultIgnoreCondition { get { throw null; } set { } }
         public bool IgnoreReadOnlyFields { get { throw null; } set { } }
         public bool IgnoreReadOnlyProperties { get { throw null; } set { } }
         public bool IgnoreRuntimeCustomConverters { get { throw null; } set { } }
         public bool IncludeFields { get { throw null; } set { } }
-        public System.Text.Json.Serialization.JsonKnownNamingPolicy NamingPolicy { get { throw null; } set { } }
+        public System.Text.Json.Serialization.JsonKnownNamingPolicy PropertyNamingPolicy { get { throw null; } set { } }
         public bool WriteIndented { get { throw null; } set { } }
+        public JsonSourceGenerationMode GenerationMode { get { throw null; } set { } }
```

Comment:
What's the semantics when you mix/match GenerationMode on `JsonSourceGenerationOptionsAttribute` and `JsonSerializableAttribute`? I assume `JsonSourceGenerationOptionsAttribute` sets it for all types, and then `JsonSerializableAttribute` overrides it for that particular type.

Reply:
Yes, that's the behavior.